File: src/com/crossbowffs/quotelock/history/provider/QuoteHistoryProvider.java
package com.crossbowffs.quotelock.history.provider;

import android.content.Context;
import android.database.sqlite.SQLiteOpenHelper;

import com.crossbowffs.quotelock.provider.AutoContentProvider;

/**
 * @author Yubyf
 */
public class QuoteHistoryProvider extends AutoContentProvider {
    public QuoteHistoryProvider() {
        super(QuoteHistoryContract.AUTHORITY, new ProviderTable[]{
                new ProviderTable(
                        QuoteHistoryContract.Histories.TABLE,
                        QuoteHistoryContract.Histories.CONTENT_ITEM_TYPE,
                        QuoteHistoryContract.Histories.CONTENT_TYPE)
        });
    }

    @Override
    protected SQLiteOpenHelper createDatabaseHelper(Context context) {
        return new QuoteHistoryHelper(context);
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/history/provider/QuoteHistoryProvider.kt
package com.crossbowffs.quotelock.history.provider

import android.content.Context
import android.database.sqlite.SQLiteOpenHelper
import com.crossbowffs.quotelock.provider.AutoContentProvider

/**
 * @author Yubyf
 */
class QuoteHistoryProvider : AutoContentProvider(
    QuoteHistoryContract.AUTHORITY, arrayOf(
        ProviderTable(
            QuoteHistoryContract.Histories.TABLE,
            QuoteHistoryContract.Histories.CONTENT_ITEM_TYPE,
            QuoteHistoryContract.Histories.CONTENT_TYPE
        )
    )
) {
    override fun createDatabaseHelper(context: Context): SQLiteOpenHelper {
        return QuoteHistoryHelper(context)
    }
}